collector/textfile.go

```diff
@@ -116,48 +241,19 @@ func (c *textFileCollector) parseTextFiles() []*dto.MetricFamily {
 		}
 	}
 
-	// Export the mtimes of the successful files.
-	if len(mtimes) > 0 {
-		mtimeMetricFamily := dto.MetricFamily{
-			Name:   proto.String("node_textfile_mtime"),
-			Help:   proto.String("Unixtime mtime of textfiles successfully read."),
-			Type:   dto.MetricType_GAUGE.Enum(),
-			Metric: []*dto.Metric{},
-		}
+	convertMetricFamilies(metricFamilies, ch)
 
-		// Sorting is needed for predictable output comparison in tests.
-		filenames := make([]string, 0, len(mtimes))
-		for filename := range mtimes {
-			filenames = append(filenames, filename)
-		}
-		sort.Strings(filenames)
+	exportMTimes(c, mtimes, ch)
 
-		for _, filename := range filenames {
-			mtimeMetricFamily.Metric = append(mtimeMetricFamily.Metric,
-				&dto.Metric{
-					Label: []*dto.LabelPair{
-						{
-							Name:  proto.String("file"),
-							Value: proto.String(filename),
-						},
-					},
-					Gauge: &dto.Gauge{Value: proto.Float64(float64(mtimes[filename].UnixNano()) / 1e9)},
-				},
-			)
-		}
-		metricFamilies = append(metricFamilies, &mtimeMetricFamily)
-	}
 	// Export if there were errors.
-	metricFamilies = append(metricFamilies, &dto.MetricFamily{
-		Name: proto.String("node_textfile_scrape_error"),
-		Help: proto.String("1 if there was an error opening or reading a file, 0 otherwise"),
-		Type: dto.MetricType_GAUGE.Enum(),
-		Metric: []*dto.Metric{
-			{
-				Gauge: &dto.Gauge{Value: &error},
-			},
-		},
-	})
-
-	return metricFamilies
+	var labels []string
```

Comment:
You don't need `labels`, just put a `nil` down below to indicate that the metric should not have any labels...